### DISCUSSION TITLE
building my first tap - why does it add a space to the command!

### DISCUSSION THREAD

**Discussion by endingwithali** [Homebrew/discussions#3588] [Tap maintenance and Homebrew development]
hi! 

working on my first tap (and also my first project in ruby) 

i've run brew create with the location of the build (it's a github release build). I'm also lightly following another tap created by my company to handle arm vs amd chips, as well as for the testing script. (the test is just to get the version to see if it downloaded using `system "#{bin}/coder", "--version"`). I ran `brew install --build-from-source <tapname>`, which succeeded, but when I try to run the actual cli tool installed, it doesnt work. What i found, after a bit of snooping is that I think it might be adding a space to the tap/command name? 

Is this due to a bad formed tap, or an issue with the source binary
![Screen Shot 2022-08-30 at 2 42 13 PM](https://user-images.githubusercontent.com/5589184/187517435-bb7d30b3-49d1-4a4e-9159-f2b6384741e8.png)

**Comment by carlocab** [comment]
> Is this due to a bad formed tap, or an issue with the source binary

The former seems more likely. But it's hard to say without actually looking at any code.

Also:

Please [don't use screenshots](https://meta.stackoverflow.com/a/285557) when sharing the output of commands in your terminal. They're often difficult to read and impossible to reuse (e.g. when looking up error messages). Instead, copy the text here and wrap them in three back-ticks (```) on the line before and on the line after to format them properly.

**Reply by endingwithali** [reply]
here's the code for the tap im trying to make 


```
class Coder < Formula
  desc "Remote development environments on your infrastructure provisioned"
  homepage "https://coder.com"
  license "AGPL-3.0-only"


  if OS.mac?
    url "https://github.com/coder/coder/releases/download/v0.8.9/coder_0.8.9_linux_amd64.tar.gz"
    sha256 "2395b06e3d39d2dbcba9f5314f972be05b4b501ebcc1c58b8fea9cdede594e60"
  else
    url "https://github.com/coder/coder/releases/download/v0.8.9/coder_0.8.9_linux_arm64.tar.gz"
    sha256 "a00413e145d872f901435f72ce343d14e80cca46b96a33919abc3f104425a12b"
  end

  def install
    bin.install "coder"
  end

  test do
    system "#{bin}/coder", "--version"
  end
end
```

**Reply by carlocab** [reply]
Is it at `Formula/coder.rb` relative to the repository root?

**Reply by SMillerDev** [reply]
Additionally, if you have pre-built binaries a cask is likely to behave much better.
